open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/JWTUtils.java

```diff
@@ -120,4 +135,101 @@ public static boolean validateJWTSignature(String jwtString, String jwksUri, Str
         return true;
     }
 
+    /**
+     *Validates the signature of a given JWT against a given public key.
+     *
+     * @param signedJWT the signed JWT to be validated
+     * @param publicKey the public key that is used for validation
+     * @param algorithm the algorithm expected to have signed the jwt
+     * @return true if signature is valid else false
+     * @throws NoSuchAlgorithmException if the given algorithm doesn't exist
+     * @throws InvalidKeySpecException if the provided key is invalid
+     * @throws JOSEException if an error occurs during the signature validation process
+     */
+    @Generated(message = "Excluding from code coverage as KeyFactory does not initialize")
+    public static boolean validateJWTSignature(SignedJWT signedJWT, String publicKey, String algorithm) throws
+            NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+
+        byte[] publicKeyData = Base64.getDecoder().decode(publicKey);
+        X509EncodedKeySpec spec = new X509EncodedKeySpec(publicKeyData);
+        KeyFactory kf = KeyFactory.getInstance(algorithm);
+        RSAPublicKey rsapublicKey = (RSAPublicKey) kf.generatePublic(spec);
+        JWSVerifier verifier = new RSASSAVerifier(rsapublicKey);
+        return signedJWT.verify(verifier);
+    }
+
+    /**
+     * Validate legitimacy of JWT.
+     *
+     * @param jwtString JWT string
+     */
+    public static boolean isJWT(String jwtString) {
```

Comment:
```suggestion
    public static boolean isJWS(String jwsString) {
```

Reply:
Fixed in [e498dd5](https://github.com/ImalshaD/financial-open-banking/pull/6/commits/e498dd5122ae34f8806ded46629fdba1c0dd5fd7)